### PR TITLE
Feature/raja hip

### PR DESCRIPTION
**Pull request by vsrana01**
ready to merge

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
Was removing the RAJA & camp submodules intentional? I think you mentioned relying on the submodules pulled in by CHAI, but if that is the case, after this PR is merged there will be no way to build Kripke without CHAI.

A better alternative would probably just to skip including the subdirectories when CHAI is enabled.

**Comment by vsrana01** [comment]
> Was removing the RAJA & camp submodules intentional? I think you mentioned relying on the submodules pulled in by CHAI, but if that is the case, after this PR is merged there will be no way to build Kripke without CHAI.
> 
> A better alternative would probably just to skip including the subdirectories when CHAI is enabled.

Ok @davidbeckingsale this can work with chai disabled. RAJA brings in camp, so I removed camp.

File: src/Kripke/Core/Field.h [hunk 1]
```diff
@@ -241,7 +244,11 @@ namespace Core {
 
         LType layout = RAJA::make_stride_one<LInfo::stride_one_dim>(m_chunk_to_layout[chunk_id]);
 
-        return ViewType<Order, ElementType, ElementPtr, IDX_TYPES...>(ptr, layout);
+#if defined(KRIPKE_USE_HIP) || defined(KRIPKE_USE_CUDA)
```
Comment (rchen20) [inline src/Kripke/Core/Field.h]: This was giving me a compilation error at `chai::GPU` in the line below.

```suggestion
#if (defined(KRIPKE_USE_HIP) || defined(KRIPKE_USE_CUDA)) && defined(KRIPKE_USE_CHAI)
```

Comment (rchen20) [reply src/Kripke/Core/Field.h]: @vsrana01 RAJA Proxy still needs this to work. If this is ok with you, would you mind committing this change? After this, I think we're good to go with merging this branch into develop.

**Review by rchen20 — APPROVED** [review]

**Pull request merged**